scipy/linalg/matfuncs.py

```diff
@@ -668,3 +675,329 @@ def rounded_sign(x):
+    Parameters
+    ----------
+    A : (N, N) array_like
+        Matrix to evaluate.
+    tol : float
+        Tolerance for the zero value.
+    
+    Returns
+    -------
+    X : boolean
+        True if a matrix A is singular.
+
+    Examples
+    --------
+    >>> from scipy.linalg import is_matrix_singular
+    >>> m = np.array([[-1,1.5],[1.5,-1]])
+    >>> print(is_matrix_singular(m))
+    True
+    >>> m = np.array([[1, -1, 0],[-1, 5, 0],[0, 0, 7]])
+    >>> print(is_matrix_singular(m))
+    True
+    """
+    A = _asarray_square(A)
+    detA = det(A)
+    return (np.absolute(detA) < tol).all()
+
+
+def is_matrix_idempotent(A, tol=1e-8):
+    """
+    Returns True if a matrix is idempotent
+    Parameters
+    ----------
+    A : (N, N) array_like
+        Matrix to evaluate.
+    tol : float
+        Tolerance for the zero value.
+    
+    Returns
+    -------
+    X : boolean
+        True if a matrix A is idempotent.
+
+    Examples
+    --------
+    >>> from scipy.linalg import is_matrix_idempotent
+    >>> m = np.array([[3,-6],[1,-2]])
+    >>> print(is_matrix_idempotent(m))
+    True
+    >>> m = np.array([[2, -2, 4],[-1, 3, 4],[1, -2, -3]])
+    >>> print(is_matrix_idempotent(m))
+    False
+    """
+    A = _asarray_square(A)
+    return (np.absolute(np.matmul(A, A) - A) < tol).all()
+
+
+def is_matrix_positive_definite(A, tol=1e-8):
+    """
+    Returns True if a matrix is positive definite
+    Parameters
+    ----------
+    A : (N, N) array_like
+        Matrix to evaluate.
+    tol : float
+        Tolerance for the zero value.
+    
+    Returns
+    -------
+    X : boolean
+        True if a matrix A is positive definite.
+
+    Examples
+    --------
+    >>> from scipy.linalg import is_matrix_positive_definite
+    >>> m = np.array([[5,1],[1,3]])
+    >>> print(is_matrix_positive_definite(m))
+    True
+    >>> m = np.array([[1, -1, 0],[-1, 5, 0],[0, 0, 7]])
+    >>> print(is_matrix_positive_definite(m))
+    True
+    """
+    A = _asarray_square(A)
+    if not is_matrix_hermitian(A):
+        raise ValueError('expected symmetric or hermitian matrix')
+    eiVal, eiVecR = eig(A)
+    eiVal[np.absolute(eiVal) < tol] = 0
+    return (eiVal > 0).all()
+
+
+def is_matrix_positive_semidefinite(A, tol=1e-8):
+    """
+    Returns True if a matrix is positive semidefinite
+    Parameters
+    ----------
+    A : (N, N) array_like
+        Matrix to evaluate.
+    tol : float
+        Tolerance for the zero value.
+    
+    Returns
+    -------
+    X : boolean
+        True if a matrix A is positive semidefinite.
+
+    Examples
+    --------
+    >>> from scipy.linalg import is_matrix_positive_semidefinite
+    >>> m = np.array([[1,-1],[-1,1]])
+    >>> print(is_matrix_positive_semidefinite(m))
+    True
+    """
+    A = _asarray_square(A)
+    if not is_matrix_hermitian(A):
+        raise ValueError('expected symmetric or hermitian matrix')
+    eiVal, eiVecR = eig(A)
+    eiVal[np.absolute(eiVal) < tol] = 0
```

Comment:
Also as we discussed in the previous PR. Identity matrix shouldn't pass this test but passes. But more importantly you have to include lots of edge cases to make this half-usable.